File: src/main/java/com/redgrapefruit/mythicaltowers/common/core/WeaponEffectTarget.java
package com.redgrapefruit.mythicaltowers.common.core;

import net.minecraft.entity.LivingEntity;

/**
 * The target of weapon effects, the target {@link LivingEntity} or the attacker {@link LivingEntity}
 */
public enum WeaponEffectTarget {
    /**
     * The target of the attack
     */
    TARGET,
    /**
     * The attacker of the attack OR the miner
     */
    ATTACKER
}

File: src/main/kotlin/com/redgrapefruit/mythicaltowers/core/WeaponEffectTarget.kt
package com.redgrapefruit.mythicaltowers.core

import net.minecraft.entity.LivingEntity

/**
 * The target of weapon effects, the target [LivingEntity] or the attacker [LivingEntity]
 */
enum class WeaponEffectTarget {
    /**
     * The target of the attack
     */
    TARGET,

    /**
     * The attacker of the attack OR the miner
     */
    ATTACKER,

    /**
     * Placeholder value
     */
    NONE
}